components/brave_rewards/browser/rewards_protocol_handler.cc

```diff
@@ -33,7 +33,38 @@ GURL TransformUrl(const GURL& url) {
                        : ""}));
 }
 
-bool IsValidWalletProviderRedirect(
+bool IsValidWalletProviderRedirect(const GURL& referrer_url,
+                                   const GURL& redirect_url) {
+  static const auto kAllowedReferrerUrls{[] {
+    std::map<std::string, std::vector<GURL>> allowed_urls{
+        {"bitflyer",
+         {GURL(BUILDFLAG(BITFLYER_PRODUCTION_URL)),
+          GURL(BUILDFLAG(BITFLYER_SANDBOX_URL))}},
+        {"gemini",
+         {GURL(BUILDFLAG(GEMINI_PRODUCTION_OAUTH_URL)),
+          GURL(BUILDFLAG(GEMINI_SANDBOX_OAUTH_URL))}},
+        {"uphold",
+         {GURL(BUILDFLAG(UPHOLD_PRODUCTION_OAUTH_URL)),
+          GURL(BUILDFLAG(UPHOLD_SANDBOX_OAUTH_URL))}},
+        {"zebpay",
+         {GURL(BUILDFLAG(ZEBPAY_PRODUCTION_OAUTH_URL)),
+          GURL(BUILDFLAG(ZEBPAY_SANDBOX_OAUTH_URL))}}};
+
+    for (const auto& [wallet_provider, urls] : allowed_urls) {
+      DCHECK(base::ranges::none_of(
+          urls,
+          [](const GURL& url) { return !url.is_valid() || !url.has_host(); }))
+          << wallet_provider << " has malformed referrer URL(s)!";
+    }
+
+    return allowed_urls;
+  }()};
+
+  return IsValidWalletProviderRedirectImpl(referrer_url, redirect_url,
+                                           kAllowedReferrerUrls);
+}
+
+bool IsValidWalletProviderRedirectImpl(
```

Comment:
using `Impl` here is not appropriate. This is not an implementation of an interface

Reply:
That was actually my suggestion. Not sure if there's a Chromium guideline that explicitly says that we can use the `Impl` postfix only in the name of a class that implements an interface. In C++, outside Chromium, it's common practice to use it elsewhere, also just a few occurrences I found quickly:
-  https://source.chromium.org/chromium/chromium/src/+/main:ipc/ipc_message_templates.h;l=26?q=function:%5Cw*Impl%20lang:cpp
- https://source.chromium.org/chromium/chromium/src/+/main:base/profiler/sample_metadata.cc;l=74?q=function:%5Cw*Impl%20lang:cpp
- https://source.chromium.org/chromium/chromium/src/+/main:ipc/ipc_message_templates.h;l=61?q=function:%5Cw*Impl%20lang:cpp

I'm sure there's more.
That said, I'm open to other names that express the intent.